### PR TITLE
fix: guard against empty component set

### PR DESCRIPTION
**Pull request by peternhale**
@W-11903179@

### CODE REVIEW

File: src/package/packageProfileApi.ts [hunk 1]
```diff
@@ -229,7 +229,7 @@ export class PackageProfileApi extends AsyncCreatable<ProfileApiOptions> {
     const profilePaths = this.findAllProfiles(excludedDirectories);
 
     // Filter all profiles
-    typesArr = typesArr.filter((kvp) => kvp.name[0] !== 'Profile');
+    typesArr = (typesArr || []).filter((kvp) => kvp.name[0] !== 'Profile');
```
Comment (peternhale) [inline src/package/packageProfileApi.ts:232]: Make sure value passed is not undefined

File: src/package/packageVersionCreate.ts [hunk 2]
```diff
@@ -110,6 +110,7 @@ export class PackageVersionCreate {
       convertResult.packagePath = outputDirectory;
       return convertResult;
     }
+    return convertResult;
```
Comment (peternhale) [inline src/package/packageVersionCreate.ts:113]: Fix return to always return the converted component set

File: src/package/packageVersionCreate.ts [hunk 3]
```diff
@@ -110,6 +110,7 @@ export class PackageVersionCreate {
       convertResult.packagePath = outputDirectory;
       return convertResult;
```
Comment (peternhale) [inline src/package/packageVersionCreate.ts:111]: I will remove this return

File: src/package/packageVersionCreate.ts [hunk 4]
```diff
@@ -419,6 +421,12 @@ export class PackageVersionCreate {
     return this.createRequestObject(preserveFiles, packageVersTmpRoot, packageVersBlobZipFile);
   }
 
+  private verifyHasSource(componentSet: ConvertResult): void {
```
Comment (peternhale) [inline src/package/packageVersionCreate.ts:424]: Verify that the component set has source present
Wrote a function so it was simple to mock

File: src/package/packageVersionCreate.ts [hunk 5]
```diff
@@ -437,6 +445,12 @@ export class PackageVersionCreate {
     const currentPackageXml = await fs.promises.readFile(path.join(packageVersMetadataFolder, 'package.xml'), 'utf8');
     // convert to json
     const packageJson = await xml2js.parseStringPromise(currentPackageXml);
+    if (!packageJson?.Package) {
```
Comment (peternhale) [inline src/package/packageVersionCreate.ts:448]: Unlikely to happen now, but leaving here just in case

**Review by shetzel — APPROVED** [review]

**Review by WillieRuemmele — APPROVED** [review]

**Pull request merged**